crates/iota-types/src/execution_status.rs

```diff
@@ -383,6 +383,80 @@ impl ExecutionStatus {
             ExecutionStatus::Failure { error, command } => (error, command),
         }
     }
+
+    /// Returns congested objects if the transaction was cancelled due to
+    /// shared object congestion, else returns `None`.
+    pub fn get_congested_objects(&self) -> Option<&CongestedObjects> {
+        if let ExecutionStatus::Failure {
+            error:
+                ExecutionFailureStatus::ExecutionCancelledDueToSharedObjectCongestion {
+                    congested_objects,
+                },
+            ..
+        } = self
+        {
+            Some(congested_objects)
+        } else if let ExecutionStatus::Failure {
+            error:
+                ExecutionFailureStatus::ExecutionCancelledDueToSharedObjectCongestionV2 {
+                    congested_objects,
+                    ..
+                },
+            ..
+        } = self
+        {
+            Some(congested_objects)
+        } else {
+            None
+        }
```

Comment:
A match would be better here I think
```suggestion
        match self {
            ExecutionStatus::Failure {
                error:
                    ExecutionFailureStatus::ExecutionCancelledDueToSharedObjectCongestion {
                        congested_objects,
                    }
                    | ExecutionFailureStatus::ExecutionCancelledDueToSharedObjectCongestionV2 {
                        congested_objects,
                        ..
                    },
                ..
            } => Some(congested_objects),
            _ => None,
        }
```